lib/rouge/lexers/liquid.rb

```diff
@@ -266,7 +266,7 @@ class Liquid < RegexLexer
         end
 
         rule %r/(with|for)\b/, Name::Tag
-        rule %r/[^\s\.]+(\.[^\s\.]+)+\b/, Name::Other
+        rule %r/[\/\w-]+(\.[\w-]+)+\b/, Text
```

Comment:
Could this be made simpler? The following will result in some false positive but will correctly match against everything that would have passed the more complicated expression, right?

```suggestion
        rule %r/[\/.\w-]+\b/, Text
```

Reply:
Oh, and at the risk of sounding silly, you wrote:

> ...marks the argument as plain `Text` to match how `include_relative` is handled.

Is `include_relative` handled by this lexer at all? I only had a quick look but it didn't look like it.

Reply:
Just checking about this as I think it's the only thing still outstanding.

Reply:
The native Liquid implementation of `include` (which is [deprecated](https://help.shopify.com/en/themes/liquid/tags/deprecated-tags#include) and undocumented except for within its [source code](https://github.com/Shopify/liquid/blob/master/lib/liquid/tags/include.rb) and [tests](https://github.com/Shopify/liquid/blob/master/test/integration/tags/include_tag_test.rb)) takes a template name, which can be a quoted string or a variable name. In the `:include` state, both of these are handled by the `:variable_tag_markup` mixin, marking them as either Literal.String or Name.Variable respectively.

The [Jekyll implementation of `include`](https://jekyllrb.com/docs/includes/) is entirely different, in that it accepts an unquoted string referring to a filename within an `_includes` folder. This rule is (re-)written to distinguish this specific form by looking for an argument that looks like a file path, i.e. a string of letters and slashes with one or more dotted extensions, and marking it as Text.

Reply:
`include_relative` is not specifically handled as it’s a Jekyll-specific tag, but it is caught by the "other tags or blocks" rule and its argument is marked as Text by the `:default_param_markup` state.

Reply:
Thanks! The change makes complete sense now :)

Reply:
Oh, one thought did occur to me: if this is for filenames, should it include backslashes as well for Windows folks?

Reply:
I don't believe that's necessary, because Ruby (and therefore Liquid and Jekyll) always uses the forward slash in paths [regardless of platform](https://stackoverflow.com/a/7173356).

Reply:
Fair point. Thanks, will merge in!